src/ints/bios.cpp

```diff
@@ -1069,14 +1070,31 @@ static void shutdown_tandy_sb_dac_callbacks()
 	tandy_dac.port = 0;
 }
 
-void BIOS_SetupTandySbDacCallbacks()
+// The Tandy Sound card requests DAC support which the following configures via
+// BIOS-based interrupt callbacks using either a Sound Blaster or the 'actual'
+// Tandy DAC module, respectively. If neither are present then the BIOS
+// callbacks aren't setup.
+//
+// The BIOS callbacks are shutdown when the backing device is shutdown to avoid
+// advertizing the DAC's presence when none exists.
+//
+// Returns true if a DAC was actually setup.
+//
+bool BIOS_ConfigureTandyDacCallbacks(const std::optional<bool> maybe_request_dac)
```

Comment:
So just to be clear, this is some DOSBox-special code right? A real Tandy with an SB card in it wouldn't do such trickery, right?

Reply:
Yup, the Tandy predated the Sound Blaster, so this deconflicting approach woulnd't have been in the original Tandy BIOS:

- Circa 2020 discussion:

    https://forum.vcfed.org/index.php?threads/is-soundblaster-compatible-with-tandy-1000-tl-2-would-plug-in-but-wont-fit-in-case.77144/

    > [mystman12](https://forum.vcfed.org/index.php?members/mystman12.40047/):  _well not everything is working perfectly. For some reason DeskMate programs that use sampled audio, like the music player and the audio recorder, crash the moment they try playing sound. I tried Life & Death and the sampled sounds still play just fine in that. I guess it's an issue specific to DeskMate?_ 

    > [Trixter](https://forum.vcfed.org/index.php?members/trixter.2251/): _The Tandy TL/2 has a built-in DAC and Deskmate is likely using that. It shouldn't crash, not sure what's going on there, but that explains why it's not trying to use the Sound Blaster._

- Circa 1996 newsgroup chat in `comp.sys.tandy` with the DeskMate author about the Tandy DAC:

    https://groups.google.com/g/comp.sys.tandy/c/7VxlbVGughE/m/wekj31wJfZYJ

- Circa 2005 document about Tandy:

    http://www.oldskool.org/guides/tvdog/1kfaq.html

    > _you can install a SoundBlaster or SoundBlaster Pro in an expansion slot. If you do so, you will have to stop using the Tandy DAC, since the SoundBlaster drivers will be confused by the DAC BIOS routines._

But given the BIOS interrupt callback is a software construct, maybe some tools or drivers were able to re-program the BIOS as a fix-up.

Also, eventually newer Sound Blaster models got better at handling this conflict using IRQ sharing at the hardware level:

    https://www.youtube.com/watch?v=jfkLDZEb8YA

    > _I finally fixed the Sound Blaster and Tandy DAC Conflict in my Tandy 1000 TL/2 Computer, and I'm beyond thrilled! the Sound Blaster Card that I used that worked is the CT1350B because it has a DMA Sharing Jumper on it, and that made all the difference!_

So given all of the above, there would have been ~1990's users able to have these hybrid Tandy Sound + Sound Blaster machines, and I suspect the DOSBox devs wanted to provide that same level of maximum compatibility.

Reply:
Clearly you've done a thorough research on this @kcgen 🎖️ Your extra comments in this PR describe the "how" well (the technicalities of this SB/Tandy DAC dance), but not the "why". I think a short top-level comment in `tandy.cpp` with an overview of this whole SB vs Tandy DAC handling business and the motivation for it would help future maintainers. Like in my case, I was just scratching my head reading your PR and the number of people aware of the background context is only going to decrease going forward 😅

Reply:
Will do :-) yes; without the logging, it was something that either worked or didn't behind the scenes!